lib/entities/enums/vote_button_type.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:pr0gramm/entities/base_types/enum.dart';
+import 'package:pr0gramm/entities/enums/vote.dart';
+
+class VoteButtonType extends Enum<int> {
```

Comment:
use a normal "enum" if possible. As far as I see you don't need the underlying int value for these types

Reply:
VoteButtonType class has method (enums cant have method) + the method is to convert to a vote so value matters a lot

Reply:
this makes it easy to compare if a btn is eq to the current vote which is used to generate the states

Reply:
Ah right, didn't catch this in the webbrowser :)